### PR TITLE
Add a comprehensive description about the use of libraries

### PR DESCRIPTION
**Pull request by SebastinSanty**

### CODE REVIEW

**Comment by prasadtalasila** [comment]
The [package.json](https://github.com/Dogfalo/materialize/blob/master/package.json) of materialize shows very few installation dependencies, only hammerjs, jquery, node-archiver. Rest of the packages are dev dependencies.

Rest looks good.

File: src/main/webapp/WEB-INF/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+## Information on the used libraries
+**Important** : All the manually-downloaded libraries which were here intially are now being installed using npm. Materialize folder is still visible here, as a download from npm, downloads all its dependencies along with the distro. This proves very costly in terms of size when war file is produced.
```
Comment (prasadtalasila) [inline src/main/webapp/WEB-INF/README.md]: how big is the war going to be? How much of this increase is due to materialize's package.json?

File: src/main/webapp/WEB-INF/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+### Libraries present:
+* `materialize`: Materialize is a modern responsive CSS framework based on Material Design by Google.
+* `backbone`: Used as the primary front-end MVC framework for this project.
+* `backbone.localstorage`: For sessions.
+* `cookie`: It is a library that helps in storing cookies.
+* `jquery`: Used by materialize and in backbone for AJAX calls and component selections.
+* `src-noconfict`: Contains ACE code editor. Used to provide an alternative to P4 file upload by providing an interface to write the P4 code on the browser.
```
Comment (prasadtalasila) [inline src/main/webapp/WEB-INF/README.md:10]: I saw some versions of the application with ACE editor in place. Now, its not there. Are there any plans of activating the editor on the P4 validation page?

Comment (SebastinSanty) [reply src/main/webapp/WEB-INF/README.md:10]: It is still there in the code. Materialize has dibs on the pre tag compared to ACE editor, hence it is not displayed. Hence, further work should override materialize defaults, to include the code editor.

**Comment by SebastinSanty** [comment]
I realized now, that the `npm install` command will install the devDependencies along other dependencies. One has to use `npm install --production`.

**Comment by prasadtalasila** [comment]
You can put in the appropriate npm instruction in [build](https://github.com/prasadtalasila/BITS-Darshini/blob/dev/scripts/build.sh) script.

File: src/main/webapp/WEB-INF/README.md (hunk 2 above)
Comment (prasadtalasila) [reply src/main/webapp/WEB-INF/README.md:10]: what is dibs?
please enable the ACE editor by default. Even with the ACE editor, the graph.p4 file upload option would still exist.

Comment (SebastinSanty) [reply src/main/webapp/WEB-INF/README.md:10]: At present, the \<pre\> tag styling is taken over by the materialize library and hence ACE code editor is not able to get hold of the \<pre\> tag. A workaround is needed in order to override this default set by materialize.

**Comment by SebastinSanty** [comment]
I'll make another PR, which should eliminate the materialize folder from the css/js folders.

**Comment by SebastinSanty** [comment]
Even `npm install --production` doesn't work. This is because the dependencies of the dependencies of materialize is not properly managed. Hence, still a lot of download is taking place.

**Comment by prasadtalasila** [comment]
Then let's keep the non-npm version of material.

**Pull request merged**